### PR TITLE
cache/reflector: check the value of the initial-events-end annotation

### PR DESCRIPTION
**Pull request by p0lyn0mial**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
we should check the value of `initial-events-end annotation` annotation so that in the future we could set it to `false`/disable.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/cache/reflector_watchlist_test.go [hunk 1]
```diff
@@ -350,6 +350,27 @@ func TestWatchList(t *testing.T) {
 			expectedStoreContent: []v1.Pod{*makePod("p1", "1"), *makePod("p3", "3")},
 			expectedError:        apierrors.NewResourceExpired("rv already expired"),
 		},
+		{
+			name:                  "prove that the reflector is checking the value of the initialEventsEnd annotation",
```
Comment (p0lyn0mial) [inline staging/src/k8s.io/client-go/tools/cache/reflector_watchlist_test.go:354]: essentially no error and an empty store

**Comment by pacoxu** [comment]
/retest
test infra failure is fixed.

**Pull request merged**